datanode/snapshot/current_snapshot.go

```diff
@@ -0,0 +1,145 @@
+package snapshot
+
+import (
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"regexp"
+	"strconv"
+)
+
+type CurrentStateSnapshot struct {
+	ChainID string
+	Height  int64
+}
+
+const currentStateSnapshotIdentifier = "currentstatesnapshot"
+
+func NewCurrentSnapshot(chainID string, height int64) CurrentStateSnapshot {
+	return CurrentStateSnapshot{
+		ChainID: chainID,
+		Height:  height,
+	}
+}
+
+func (s CurrentStateSnapshot) UncompressedDataDir() string {
+	return fmt.Sprintf("%s-%d-%s", s.ChainID, s.Height, currentStateSnapshotIdentifier)
+}
+
+func (s CurrentStateSnapshot) CompressedFileName() string {
+	return fmt.Sprintf("%s-%d-%s.tar.gz", s.ChainID, s.Height, currentStateSnapshotIdentifier)
+}
+
+func (s CurrentStateSnapshot) String() string {
+	return fmt.Sprintf("{Current State Snapshot Chain ID:%s Height:%d}", s.ChainID, s.Height)
+}
+
+func (s CurrentStateSnapshot) GetCopySQL(dbMetaData DatabaseMetadata, databaseSnapshotsPath string) []string {
+	var copySQL []string
+	for tableName, meta := range dbMetaData.TableNameToMetaData {
+		if !dbMetaData.TableNameToMetaData[tableName].Hypertable {
+			snapshotFile := filepath.Join(databaseSnapshotsPath, s.UncompressedDataDir(), tableName)
+			tableCopySQL := fmt.Sprintf(`copy (select * from %s order by %s) to '%s'`, tableName,
+				meta.SortOrder, snapshotFile)
+			copySQL = append(copySQL, tableCopySQL)
+		}
+	}
+
+	return copySQL
+}
+
+func FileExists(snapshotsDir string, file string) (bool, error) {
```

Comment:
This is essentially [this](https://github.com/vegaprotocol/vega/blob/develop/libs/fs/fs.go#L39), so we should reuse.